PKGBUILD

```diff
@@ -37,24 +38,35 @@ optdepends=('quartus-lite-max: MAXII, MAXV device support'
            # Pkgnames are taken directly from altera downloads file names
 
 source=("http://download.altera.com/akdlm/software/acdsinst/${pkgver%.*.*}std.1/${_build_nr}/update/QuartusSetup-${pkgver}-linux.run"
-  "http://download.altera.com/akdlm/software/acdsinst/${pkgverbase%.*.*}std/${_build_nr_base}/ib_installers/QuartusLiteSetup-${pkgverbase}-linux.run"
-	"quartus.desktop" "51-usbblaster.rules" "quartus.install")
+    "http://download.altera.com/akdlm/software/acdsinst/${pkgverbase%.*.*}std/${_build_nr_base}/ib_installers/QuartusLiteSetup-${pkgverbase}-linux.run"
+    "quartus.desktop" "51-usbblaster.rules" "quartus.install")
 md5sums=('70e8016ea12cf7835dfcd3b22b1e3153'
          '8a22e65f15b695e7967a292caa7275f3'
          'd7181c4c6d88c7bd34061214ee350d73'
          'f5744dc4820725b93917e3a24df13da9'
          'a331a81c44aed062a7af6d28542c3d82')
 
-#options=('strip' 'upx') # Stripping and UPX will takes ages, I'd avoid it.
+options=('!strip' '!upx') # Stripping and UPX will takes ages, I'd avoid it.
 install='quartus.install'
 PKGEXT=".pkg.tar" # Do not compress
 
-package() {
+prepare() {
     cd "${srcdir}"
 
-    # TODO: Make bogus $DISPLAY
     chmod a+x "QuartusLiteSetup-${pkgverbase}-linux.run" "QuartusSetup-${pkgver}-linux.run"
-    DISPLAY="" ./"QuartusLiteSetup-${pkgverbase}-linux.run" --mode unattended --unattendedmodeui none --accept_eula 1 --installdir "${pkgdir}/${_alteradir}"
+}
+
+package() {
+    cd "${srcdir}"
+
+    # FIXME: Create log dir for the following workaround
+    mkdir -p "${pkgdir}${_alteradir}/logs"
+
+    DISPLAY="" ./"QuartusLiteSetup-${pkgverbase}-linux.run" --mode unattended \
+        --unattendedmodeui none --accept_eula 1 --installdir "${pkgdir}/${_alteradir}" &
+    # FIXME: Installer doesn't finish, as a workaround kill it as soon as the install log is created
+    inotifywait -qq -e create "${pkgdir}${_alteradir}/logs" &&
+    killall "QuartusLiteSetup-${pkgverbase}-linux.run"
```

Comment:
Doesn't killing return an error code and cause makepkg to terminate? That's what I was happening when I killed it. Or does the 'backgrounding' & fix that?

Also, any idea why it actually fails to terminate on its own?

Reply:
It doesn't cause the makepkg to terminate when it is killed, exactlly for the reason you guessed, because it is send to background. It's the best workaround I could think of, it's still ugly. Before, I was appending `|| true` to the installer and killing it by hand.

Unfortunettly I have no idea why it failes to terminat.